clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -983,10 +983,13 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
 
   if (getLangOpts().SYCLIsDevice && D) {
     if (const auto *A = D->getAttr<SYCLIntelLoopFuseAttr>()) {
-      Expr *E = A->getValue();
+      const Expr *Arg = A->getValue();
+      const auto *CE = cast<ConstantExpr>(Arg);
+      assert(CE && "Not an integer constant expression");
```

Comment:
You can remove this assertion -- `cast<>` can never return null (it either returns a valid object or it asserts).

Reply:
Yes, right. I missed that. Thanks for pointing this. Removed the assert.